app/stats/views.py

```diff
@@ -1,88 +1,98 @@
-import altair as alt
-from django.http import JsonResponse
-from django.shortcuts import render
+from datetime import date
+
 import altair
 import duckdb
 import pandas as pd
-from datetime import date
+import urllib
+from django.http import HttpResponseRedirect, JsonResponse
+from django.shortcuts import render
+from django.views.decorators.csrf import csrf_exempt
+
+from .forms import StatFilterForm
+
 
 def retrieve_graph_data(request):
     scope = request.GET.get('scope')
-    # tld = request.GET.get('tld')
+    tld = request.GET.get('tld')
 
     if (scope == 'monthly'):
-        data = retrieve_monthly_chart()
+        data = retrieve_monthly_chart(tld)
     else:
-        data = retrieve_annual_chart()
+        data = retrieve_annual_chart(tld)
     
     return JsonResponse(data.to_dict(), safe = False)
 
+@csrf_exempt
 def index(request):
-    return render(request, 'index.html')
+    if request.htmx:
+        # Refresh partial template
+        form = StatFilterForm(request.POST or None)
+
+        if form.is_valid():
+            time_scope = form.cleaned_data.get("time_scope")
+            tld = form.cleaned_data.get("tld")
 
-def retrieve_annual_chart(from_year = 2009, to_year = date.today().year):
-    from_year = str(from_year)
-    to_year = str(to_year)
+        template_name = "partials/greencheck_graph.html"
+        graph_data_url = "/retrieve_graph_data/?" + urllib.parse.urlencode({'scope': time_scope,'tld': tld})
```

Comment:
Would you mind using an f-string here instead? I think they're easier to read

https://realpython.com/python-f-strings/